src/api/baseapi.cpp

```diff
@@ -1903,7 +1909,9 @@ void TessBaseAPI::End() {
   delete block_list_;
   block_list_ = nullptr;
   if (paragraph_models_ != nullptr) {
-    paragraph_models_->delete_data_pointers();
+    for (auto model : *paragraph_models_) {
+      delete model;
```

Comment:
Not good.
We should consider unique ptrs or something like this here.
I also tried to replace `delete_data_pointers()`, but noticed that it frees memory of elements and reverted changes locally.

There are several common patterns over the codebase. I think those places must be upgraded uniformely.

Reply:
My first priority is currently getting a fully working git master again. A lot of code can be simplified or rewritten, but that can be done later.